src/tools/clippy/tests/ui/transmute_int_to_non_zero.stderr

```diff
@@ -2,63 +2,63 @@ error: transmute from a `u8` to a `NonZeroU8`
   --> $DIR/transmute_int_to_non_zero.rs:18:33
    |
 LL |     let _: NonZeroU8 = unsafe { std::mem::transmute(int_u8) };
-   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `NonZeroU8::new_unchecked(int_u8)`
+   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using: `NonZero::new_unchecked(int_u8)`
```

Comment:
Since this feature will be unstable, we should not suggest `NonZero` without gating on the feature in Clippy. This test file should stay unchanged, until `NonZero` is stabilized. You might want to add another test file with the feature enabled to test, if the generic `NonZero::...` suggestion is produced, but that's optional.

Reply:
1. How to gate on the feature?
2. Alternatively, is there any way of detecting whether a `rust_middle::ty::Ty` is a type alias?

Reply:
You can check over the `cx.sess().features.<feature_name>` whether the feature is enabled or not.

You can probably check for the `ty.kind() == ty::Alias(..)`: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_type_ir/sty/enum.TyKind.html#variant.Alias